stratum/docs/setup_guide_barefoot_tofino_onos.md

```diff
@@ -0,0 +1,211 @@
+```
+
+This will start Stratum in the foreground (not detached).
+
+:warning: **Important**: Huge table allocation **must** be activated on the host system, see [here](https://github.com/stratum/stratum/blob/master/stratum/hal/bin/barefoot/README.md#huge-pages--dma-allocation-error).
+
+:pencil2: (Optional) You can tunnel the ports exposed by Stratum (P4Runtime, gNMI, gNOI) via SSH tunnels:
+
+```
+for port in 9339 28000; do
+    ssh \
+        -o ExitOnForwardFailure=yes \
+        -f \
+        -N \
+        -L 0.0.0.0:$port:localhost:$port \
+        $HOST
+done
+```
+The ports on the Stratum switch are now tunneled to your local ports, e.g., accessing localhost:9339 will actually access port 9339 on the switch.
+
+
+## Start ONOS
+
+ONOS can be started on any host with a connection to switch.
+
+```
+docker run \
+    --rm \
+    --tty \
+    --detach \
+    --publish 8181:8181 \
+    --publish 8101:8101 \
+    --publish 5005:5005 \
+    --publish 830:830 \
+    --name onos \
+    onosproject/onos:2.2.3
+
+docker logs --follow onos
+```
+
+Visit [http://localhost:8181/onos/ui/login.html](http://localhost:8181/onos/ui/login.html) to confirm that ONOS is initialized (can take some while).
+
+:pencil2: The `user:password` for ONOS is `onos:rocks`.
+
+:pencil2: (Optional) For easier handling, you can tunnel the ONOS ports to your local machine:
+```
+HOST="THE_SSH_HOST_YOU_STARTED_ONOS_ON"
+for port in 8181 8101 5005 830; do
+    ssh \
+        -o ExitOnForwardFailure=yes \
+        -f \
+        -N \
+        -L 0.0.0.0:$port:localhost:$port \
+        $HOST
+done
+```
+This tunnels the ports to your local machine which enables you to execute ONOS commands locally, e.g., `onos onos@localhost` (user/password is the same as for the GUI).
+
+
+
+## Enable ONOS apps
+
+Enter the ONOS CLI (`onos onos@localhost`) and execute
+```
+onos> app activate org.onosproject.drivers.barefoot
+onos> app activate org.onosproject.segmentrouting
+```
+
+## Compile `fabric-tofino`
+
+[Main page](https://github.com/opencord/fabric-tofino)
+
+You now have to install a ONOS pipeconf for the Tofino switch.
+
+```
+git clone git@github.com:opencord/fabric-tofino.git
+cd fabric-tofino
+
+# This will take a while - generated a *.oar file containing the Pipeconfs with compiled P4 programs (drivers for ONOS)
+make pipeconf
+# This will install the app (*.oar) in ONOS. It will be activated as well
+make pipeconf-install ONOS_HOST=localhost
+```
+
+## Connect ONOS to your switch
+
+Adapt the `tofino-netcfg.json` in the `fabric-tofino` to your switch. See [fabric-tofino README](https://github.com/opencord/fabric-tofino#4---connect-onos-to-a-stratum-switch) for more information.
+
+In particular, you have to adapt the `pipeconf` part in the `tofino-netcfg.json` to your switch.
+
+There are two codenames for Tofino ASICs, namely: `montara` (Tofino 18D/20D/32D), and `mavericks` (Tofino 32Q/64Q).
+The Accton Wedge 100BF-32X is powered by `montara`.
```

Comment:
This is not really correct. `montara` and `mavericks` are Edegcore internal names. Since they were the first ones with the Tofino Wedge, the name got adopted by a part of the community. Now, with many vendors, this naming is misleading.

I struggle to find a better wording for this section, so up to you.

Also, if you want to be super useful, tell the users how to find out which chip they have:
```
bf-sde> pipe_mgr dev
-------------------------------------
Device|Type |#pipe|#stg|#prsr|#macblk
------|-----|-----|----|-----|-------
0     |T32D |2    |12  |18   |-1
```

Reply:
Thanks for the suggestion! I will change it accordingly.

Regarding the SDE shell: I wrote this guide to circumvent needing the SDE since I do not have access to it, yet. Strangely enough, Barefoot did not answer my request for access.

But I will add a notice for the user.

Reply:
I added your explanation to the README!

Reply:
The shell becomes accessible by setting this flag: `-bf_switchd_background=false`.

Reply:
As I do not have access to the SDE, I do not know where one has to apply this flag! If you say which command has to be executed (and where the flag has to be passed), I will add it.

Reply:
Should have been more specific:

Let's say this is how you run Stratum now: `start-stratum-container.sh`
It would become: `start-stratum-container.sh -bf_switchd_background=false`
Also try `-help` for a list of all commands.

Reply:
@pudelkoM Thanks - I added it.